File: Homework/Ngan_Luu/513-find-bottom-left-tree-value.py
# 513. https://leetcode.com/problems/find-bottom-left-tree-value/
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        queue=[root]
        while len(queue)>0:
            node=queue[0]
            queue.pop(0)
            if node.right:
                queue.append(node.right)
            if node.left:
                queue.append(node.left)
        return node.val

File: Homework/Hoang_Nguyen/lesson7_2/513_findBottomLeftTreeValue.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var findBottomLeftValue = function (root) {
  const queue = [];
  queue.unshift(root);
  let node;

  while (queue.length) {
    node = queue.pop();
    if (node) {
      if (node.right) {
        queue.unshift(node.right);
      }
      if (node.left) {
        queue.unshift(node.left);
      }
    }
  }
  return node.val;
};
